pandas/core/indexes/category.py

```diff
@@ -365,79 +364,6 @@ def __contains__(self, key: Any) -> bool:
 
         return contains(self, key, container=self._engine)
 
-    # TODO(2.0): remove reindex once non-unique deprecation is enforced
-    def reindex(
-        self, target, method=None, level=None, limit=None, tolerance=None
-    ) -> tuple[Index, npt.NDArray[np.intp] | None]:
-        """
-        Create index with target's values (move/add/delete values as necessary)
-
-        Returns
-        -------
-        new_index : pd.Index
-            Resulting index
-        indexer : np.ndarray[np.intp] or None
-            Indices of output values in original index
-
-        """
-        if method is not None:
-            raise NotImplementedError(
-                "argument method is not implemented for CategoricalIndex.reindex"
-            )
-        if level is not None:
-            raise NotImplementedError(
-                "argument level is not implemented for CategoricalIndex.reindex"
-            )
-        if limit is not None:
-            raise NotImplementedError(
-                "argument limit is not implemented for CategoricalIndex.reindex"
-            )
-
```

Comment:
suggestion: keep the checks above, and then here do `return super().reindex(target)`

Reply:
Sure, updated per your suggestion.